JumpscaleLibs/sal/zosv2/Zosv2.py

```diff
@@ -0,0 +1,197 @@
+        peer = nr1.peers.new()
+        peer.iprange = network2
+
+        peer.allowed_iprange = ["100.64." + str(allowed_ip[1]) + "." + str(allowed_ip[2]) + "/32", network2]
+        # this is your wireguard public key from your laptop
+        peer.public_key = public_key.decode()
+
+        self.config.public_key = public_key
+        self.config.private_key = private_key
+        self.config.wg_port = wg_port
+
+        return network, self.config
+
+    def add_volume(self,reservation, node_id, voulme_size=5, volume_type="SSD"):
+        volume = reservation.data_reservation.volumes.new()
+        volume.workload_id = 2
+        volume.size = voulme_size
+        volume.type = volume_type
+        volume.node_id = node_id
+        return  volume
+
+    def get_all_ips(self, ip_range):
+        networks = netaddr.IPNetwork(ip_range)
+        ips = []
+        for ip in list(networks.iter_hosts()):
+           ips.append(ip.format())
+        return ips
+
+    def create_container(self,network_name,ip_address,node_id, reservation, flist, env={}, entrypoint="",interactive=False, volume_size=0, voulme_type="SSD", mount_point="/sandbox/var"):
+
+        cont = reservation.data_reservation.containers.new()
+        cont.node_id = node_id
+        cont.workload_id = 3
+
+        cont.flist = flist
+        cont.storage_url = "zdb://hub.grid.tf:9900"
+        cont.environment = env
+        cont.entrypoint = entrypoint
+        cont.interactive = interactive
+
+        net = cont.network_connection.new()
+        net.network_id = network_name
+        net.ipaddress = ip_address
+
+        if volume_size > 0:
+            volume = self.add_volume(reservation,node_id,volume_size,voulme_type)
+            vol = cont.volumes.new()
+            # here we reference the volume created in the same reservation
+            vol.workload_id = 3
+            vol.volume_id = f"-{volume.workload_id}"
+            vol.mountpoint = mount_point
+
+        return reservation
+
+
+    def wg_config(self, ip_range,network_config):
+
+        first_network_range = netaddr.IPNetwork(ip_range).ip
+        ip = netaddr.IPAddress(first_network_range)
+        first_ip = ip.words
+        # add 256 + 256 to 172.20.0.0 -> 172.20.2.0
+        first_network_range += 512
+        ip = netaddr.IPAddress(first_network_range)
+        allowed_ip = ip.words
+
+        result = f"""
+        [Interface]
+        Address = 100.64.{allowed_ip[1]}.{allowed_ip[2]}/16, {first_network_range}/16
+        PrivateKey = {network_config.private_key.decode()}
+        [Peer]
+        PublicKey = {network_config.public_key.decode()}
+        Endpoint = [{network_config.public_ip}]:{network_config.wg_port}
+        AllowedIPs = {ip_range}, 100.64.{first_ip[1]}.{first_ip[2]}/16
+        PersistentKeepalive = 25"""
+        return result
+
+    def register(self, reservation):
+        reservation.json = reservation.data_reservation._json
+        me = j.tools.threebot.me.default
```

Comment:
this should probably be an argument. Since a instance of jumspcale can host multiple instance of `j.tools.threebot.me` the user should be able to choose which one he wants to use to sign,

Reply:
can please explain that, i don't understand what you need exactly ?
because it's just workaround till to solve the problem of signature